client/human/HumanClientFSM.cpp

```diff
@@ -99,19 +99,27 @@ WaitingForSPHostAck::~WaitingForSPHostAck()
 boost::statechart::result WaitingForSPHostAck::react(const HostSPGame& msg) {
     if (TRACE_EXECUTION) DebugLogger() << "(HumanClientFSM) WaitingForSPHostAck.HostSPGame";
 
-    Client().Networking().SetPlayerID(msg.m_message.ReceivingPlayer());
-    Client().Networking().SetHostPlayerID(msg.m_message.ReceivingPlayer());
+    try {
+        int host_id = boost::lexical_cast<int>(msg.m_message.Text());
 
-    Client().GetClientUI().GetMapWnd()->Sanitize();
+        Client().Networking().SetPlayerID(host_id);
+        Client().Networking().SetHostPlayerID(host_id);
 
-    return transit<PlayingGame>();
+        Client().GetClientUI().GetMapWnd()->Sanitize();
+
+        return transit<PlayingGame>();
+    } catch (...) {
+        ErrorLogger() << "WaitingForSPHostAck::react(const HostSPGame& msg) couldn't parese message text: " << msg.m_message.Text();
```

Comment:
Spelling parese -> parse.

And I'd use a message like

`"Host id " << msg.m_message.Text() << " is not a number."`